src/spectrecoff/Calendar.fs

```diff
@@ -0,0 +1,83 @@
+[<AutoOpen>]
+module SpectreCoff.Calendar
+
+open Spectre.Console
+open SpectreCoff.Layout
+open SpectreCoff.Output
+
+type Culture = Culture of string
+let unwrapCulure (Culture c) = c
+
+type Year = Year of int
+[<RequireQualifiedAccess>]
+module Year = 
+    let unwrap (Year y) = y
+
+type Month = Month of int
+[<RequireQualifiedAccess>]
+module Month = 
+    let unwrap (Month m) = m
+
+type Day = Day of int
+[<RequireQualifiedAccess>]
+module Day = 
+    let unwrap (Day d) = d
+
+type Event = Event of Year * Month * Day
+[<RequireQualifiedAccess>]
+module Event =
+    let toTuple event = 
+        match event with
+        | Event (y,m,d) -> (Year.unwrap y, Month.unwrap m, Day.unwrap d)
+
+type CalendarSettings = 
+    {  Culture: Culture Option;
+       HideHeaders: bool;
+       HeaderColor: Color;
+       HeaderStyle: Decoration;
+       HighlightColor: Color Option
+       HighlightStyle: Decoration Option }
+
+let defaultCalendarSettings = 
+    {  Culture =  None
+       HideHeaders = false
+       HeaderColor = calmColor
+       HeaderStyle = Decoration.Bold
+       HighlightColor = Some pumpedColor
+       HighlightStyle = Some Decoration.Invert }
+
+let toOutputPayload calendar =
+    calendar 
+    :> Rendering.IRenderable 
+    |> Renderable
+
+let addEvent event (calendar: Calendar) =
+    event 
+    |> Event.toTuple
+    |> calendar.AddCalendarEvent
+
+let applysettings (settings: CalendarSettings) (calendar: Calendar) = 
+    match settings.Culture with 
+    | Some culture -> calendar.Culture <- unwrapCulure culture |> System.Globalization.CultureInfo
+    | _ -> ()
+
+    calendar.ShowHeader <- not settings.HideHeaders
```

Comment:
Why not have our settings also have `ShowHeaders`? It matches `SpectreConsole`'s api and this double negation is slightly confusing.
Is it that bool defaults to false and therefore headers will be shown by default?

Reply:
The Spectre.Console api is a bit confusing there. If you look at the calendar (or table, for that matter) documentation, you'll see this example:

<img width="257" alt="image" src="https://user-images.githubusercontent.com/10219667/212471428-fbb1db18-5795-495e-8068-5477507a555b.png">

But the underlying property you can set on the class is called Show*. And yes, since bools default to false, and the default should be to show the header, I think it makes sense this way. We did the same for tables previously, and internal consistency in _SpectreCoff_ is to be respected as well.

Reply:
So to be more clear: `HideHeader()` is an extension method on the calendar class, `ShowHeader `a boolean property. We need to set the property, but in practice, I think the extension method will be used more often.

Reply:
Hmm yes got it thanks